ndbench-es-plugins/src/main/java/com/netflix/ndbench/plugin/es/EsWriter.java

```diff
@@ -0,0 +1,190 @@
+package com.netflix.ndbench.plugin.es;
+
+import com.google.common.collect.ImmutableMap;
+import com.netflix.ndbench.api.plugin.DataGenerator;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.message.BasicHeader;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.RestClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+import java.util.UUID;
+
+
+/**
+ * <p>
+ * Writes instances of JSON documents returned by {@link EsUtils#createDefaultDocumentAsJson} to Elasticsearch either
+ * via single  REST calls, or via the bulk write API.
+ * <p>
+ * The responsibilities of this class are to:
+ * <uL>
+ * <li>
+ * determine if a particular write operation is done via Elasticsearch's bulk API or not (using the value of
+ * the isBulkWrite  boolean passed to this class's constructor)
+ * </li>
+ * <p>
+ * <li>
+ * immediately issue a PUT of the document to Elasticsearch if a given write is determined not to be handled by
+ * bulk write.
+ * </li>
+ * <p>
+ * <li>
+ * maintain a thread local buffer of pending documents to be bulk written.
+ * </li>
+ * <p>
+ * <li>
+ * flush the pending buffer once its size equals {@link IEsConfig#getBulkWriteBatchSize()}.
+ * </li>
+ * <p>
+ * </uL>
+ * </p>
+ * Note that if a document is written via bulk write its isBulkWrite attribute will be "true".
+ * <br/>
+ * Note that any exception thrown will be bubbled up to the driver and will result in a failed write being recorded.
+ * <p>
+ */
+class EsWriter {
+    private static final Logger logger = LoggerFactory.getLogger(EsWriter.class);
+
+    private final String esIndexUrl;
+    private final int bulkWriteBatchSize;
+    private final String indexName;
+    private final String esDocType;
+    private final boolean isBulkWrite;
+
+    private static final ImmutableMap<String, String> EMPTY_PARAMS = ImmutableMap.of();
```

Comment:
Instead of having this variable, you can just use `Collections.emptyMap()` from `java.util` which is empty, immutable, static, and final.